cairo/ethereum/cancun/state.cairo

```diff
@@ -1403,3 +1381,67 @@ func mapping_address_bytes32_write{range_check_ptr, mapping: MappingAddressBytes
     );
     return ();
 }
+
+func finalize_state{range_check_ptr, state: State}() {
+    alloc_locals;
+    // Squash the main trie for unique keys
+    let main_trie = state.value._main_trie;
+    let main_trie_start = cast(main_trie.value._data.value.dict_ptr_start, DictAccess*);
+    let main_trie_end = cast(main_trie.value._data.value.dict_ptr, DictAccess*);
+
+    let (squashed_main_trie_start, squashed_main_trie_end) = default_dict_finalize(
+        main_trie_start, main_trie_end, cast(main_trie.value.default.value, felt)
+    );
+
+    tempvar squashed_main_trie = TrieAddressOptionalAccount(
+        new TrieAddressOptionalAccountStruct(
+            secured=bool(1),
```

Comment:
why do we hardcode values here but below in the storage tries we use the storage_tries attributes `.secured` and `.default`